pkg/apis/certmanager/validation/certificate_for_issuer_test.go

```diff
@@ -0,0 +1,62 @@
+package validation
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha1"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func TestValidateCertificateForACMEIssuer(t *testing.T) {
+	fldPath := field.NewPath("spec")
+	scenarios := map[string]struct {
+		spec   *v1alpha1.CertificateSpec
+		issuer *v1alpha1.IssuerSpec
+		errs   []*field.Error
+	}{
+		"valid basic certificate": {
+			spec: &v1alpha1.CertificateSpec{
+				CommonName: "testcn",
+				SecretName: "abc",
+				IssuerRef:  validIssuerRef,
+			},
+			issuer: &v1alpha1.IssuerSpec{},
+		},
+		"certificate with invalid keyAlgorithm": {
+			spec: &v1alpha1.CertificateSpec{
+				CommonName:   "testcn",
+				SecretName:   "abc",
+				IssuerRef:    validIssuerRef,
+				KeyAlgorithm: v1alpha1.KeyAlgorithm("blah"),
+			},
+			errs: []*field.Error{
+				field.Invalid(fldPath.Child("keyAlgorithm"), v1alpha1.KeyAlgorithm("blah"), "ACME key algorithm must be RSA"),
+			},
+		},
+		"certificate with correct keyAlgorithm for ACME": {
+			spec: &v1alpha1.CertificateSpec{
+				CommonName:   "testcn",
+				SecretName:   "abc",
+				IssuerRef:    validIssuerRef,
+				KeyAlgorithm: v1alpha1.RSAKeyAlgorithm,
+			},
+		},
```

Comment:
Can we add a case for a valid type, but invalid for this issuer? e.g. setting key algorithm to ecdsa

Reply:
:+1: